### ISSUE TITLE
g2gen: clippy says "suspicious use of `^` in `Add` impl"

### ISSUE THREAD

**Issue by KOBA789** [WanzenBug/g2p#5]
First of all, great thanks for the awesome library!

I use clippy(0.1.66 (69f9c33 2022-12-12)) and I got:
```
warning: suspicious use of `^` in `Add` impl
 --> src/main.rs:3:1
  |
3 | / g2p::g2p!(
4 | |     GF256,
5 | |     8,
6 | |     modulus: 0b1_1000_0111,
7 | | );
  | |_^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#suspicious_arithmetic_impl
  = note: `#[warn(clippy::suspicious_arithmetic_impl)]` on by default
  = note: this warning originates in the macro `g2p::g2p` (in Nightly builds, run with -Z macro-backtrace for more info)
```

I think that it has no problems to use`^` to implement galois field, of cource.
However, it seems that clippy can't understand it.

The only problem is that there is no reasonable way to suppress this warning on the crate user side.

No effects (unfortunately):
```
#[warn(clippy::suspicious_arithmetic_impl)]
g2p::g2p!(GF256, 8, modulus: ...)
```

This works, but it's excessive:
```
#![allow(clippy::suspicious_arithmetic_impl)] // at the first line of module
```

Could you add some workarounds? (it's clippy's matter I think too but...)

**Comment by WanzenBug** [comment]
> ```
> #[warn(clippy::suspicious_arithmetic_impl)]
> g2p::g2p!(GF256, 8, modulus: ...)
> ```

Shouldn't this be
```
#[allow(clippy::suspicious_arithmetic_impl)]
g2p::g2p!(GF256, 8, modulus: ...)
```
?

**Comment by WanzenBug** [comment]
In any case, just released 1.0.1 which has the right annotations to disable clippy where required
